utils/manage_events.py

```diff
@@ -0,0 +1,103 @@
+from ha.core.event_manager.event_manager import EventManager
+
+COMPONENT_ID = 'hare'
+
+class EventListener():
+    """This is listener for event manager
+
+    Typical use case of this class is as follows
+    subscribe(event list) - subscribe to events for listening, returns topic
+                            which can be used for listening
+    _listen(topic) - reading messages received on above topic
+    unsubscribe(event list) - unsubscribe as we dont want to receive these
+                              events anymore
+    get_message_type() - returns topic just in case user is not maintaining
+                         output of 'subscribe'
+    Note: 1. There is only 1 topic(message type, queue) for given component.
+          2. When component subcribes for event for the 1st time topic is
+             created and returned to user.
+          3. This topic will be used by user to listen for events.
+          4. After event is received in queue, user will read it, process it
+             and acknowledge it
+
+    """
+
+    event_manager = None
+    consumer = None
+
+    def __init__(self):
+        logging.debug('Inside EventListener')
+        self.event_manager = EventManager.get_instance()
+
+    def subscribe(self, event_list: List[str]) -> str:
+        logging.info('Subscribing for events: (%s)', str(event_list)[1:-1])
+        kafka_topic_name = self.event_manager.subscribe(COMPONENT_ID, event_list)
+        return kafka_topic_name
+
+    # Gets message type name mapped with component name
+    # Event manager creates message type (queue/topic) when component
+    # subscribes the event for the first time
+    def get_message_type(self) -> str:
+        logging.debug('get_message_type')
+        return self.event_manager.message_type(COMPONENT_ID)
+
+    # Get list of events subscribed by Hare
+    def get_events(self) -> List[str]:
+        logging.debug('Get list')
+        return self.event_manager.get_events(COMPONENT_ID)
+
+    def unsubscribe(self, event_list: List[str]) -> bool:
+        logging.info('Unsubscribing for events: (%s)', str(event_list)[1:-1])
+        return self.event_manager.unsubscribe(COMPONENT_ID, event_list)
+
+    # Listen for events from event manager
+    # Once user gets message using below command user needs to call ack() to
+    # acknowledge that message is already processed
+    def listen(self, topic: str, time_out):
+        logging.info('Listening on topic: (%s)', topic)
+        if self.consumer is None:
+            message_bus = MessageBus()
+            self.consumer = MessageConsumer(message_bus=message_bus,
+                                            consumer_id=COMPONENT_ID,
+                                            consumer_group=COMPONENT_ID,
+                                            message_types=[topic],
+                                            auto_ack=False,
+                                            offset='earliest')
```

Comment:
I'm a bit confused that this consumer is created every time the user invokes this `listen` method.

Reply:
No, it will be created only once.
Below check will make sure that same consumer will be used 2nd time onwards.
`if self.consumer is None:`

Reply:
Sorry, you're right.